### PR TITLE
Add the missing dockerimage component endpoint attributes to the devfile reference

### PR DESCRIPTION
**Pull request by metlos**
### What does this PR do?
`$TITLE`

### What issues does this PR fix or reference?
I've discovered this while investigating the docs for workspace exposure strategies (https://github.com/eclipse/che/issues/14189)

### CODE REVIEW

**Review by skabashnyuk — APPROVED** [review]

File: src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc [hunk 1]
```diff
@@ -386,6 +386,14 @@ Here, there are two dockerimages, each defining a single endpoint. Endpoint is a
 
 * `protocol`: For public endpoints the protocol is a hint to the UI on how to construct the URL for the endpoint access. Typical values are `http`, `https`, `ws`, `wss`.
 
+* `secure`: A boolean (defaulting to `false`) specifying whether the endpoint will be put behind a JWT proxy requiring a JWT workspace token to grant access.
+
+* `path`: The URL path of the endpoint
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc]:
```suggestion
* `path`: The URL of the endpoint
```

File: src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc [hunk 2]
```diff
@@ -386,6 +386,14 @@ Here, there are two dockerimages, each defining a single endpoint. Endpoint is a
 
 * `protocol`: For public endpoints the protocol is a hint to the UI on how to construct the URL for the endpoint access. Typical values are `http`, `https`, `ws`, `wss`.
 
+* `secure`: A boolean (defaulting to `false`) specifying whether the endpoint will be put behind a JWT proxy requiring a JWT workspace token to grant access.
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc]:
```suggestion
* `secure`: A boolean (defaulting to `false`) specifying whether the endpoint is put behind a JWT proxy requiring a JWT workspace token to grant access.
```

File: src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc [hunk 3]
```diff
@@ -386,6 +386,14 @@ Here, there are two dockerimages, each defining a single endpoint. Endpoint is a
...
+
+* `path`: The URL path of the endpoint
+
+* `unsecuredPaths`: A comma-separated list of paths in the endpoint that should not be secured, even if the `secure` attribute is set to `true`
+
+* `cookieAuthEnabled`: If set to `true` (the default is `false`), the JWT workspace token will be automatically fetched and put to a workspace-specific cookie so that the JWT proxy let's the requests through. Note that this brings along with it a potential danger of a link:https://en.wikipedia.org/wiki/Cross-site_request_forgery[CSRF] attack if used in conjunction with a server using POST requests. Use this after a careful consideration.
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc]:
```suggestion
* `cookieAuthEnabled`: When set to `true` (the default is `false`), the JWT workspace token is automatically fetched and included in a workspace-specific cookie to allow requests to pass through the JWT proxy.
+
WARNING: This setting potentially allows a link:https://en.wikipedia.org/wiki/Cross-site_request_forgery[CSRF] attack when used in conjunction with a server using POST requests.

File: src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc [hunk 4]
```diff
@@ -386,6 +386,14 @@ Here, there are two dockerimages, each defining a single endpoint. Endpoint is a
...
+
 If you start a new server within your component, Che will autodetect this and the UI will offer you to automatically expose this port as a `public` port. This is useful for debugging a web application, for example. But, it is not possible to do this for servers that autostart with the container (for example, a database server). For such components, you must specify the endpoints explicitly.
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc]:
```suggestion
When starting a new server within a component, Che autodetects this, and the UI offers to automatically expose this port as a `public` port. This is useful for debugging a web application, for example. It is not possible to do this for servers that autostart with the container (for example, a database server). For such components, specify the endpoints explicitly.
```

**Review by rkratky — CHANGES_REQUESTED** [review]

**Review by rkratky — APPROVED** [review]

**Comment by themr0c** [comment]
Hi @metlos. The changes that you have committed from GitHub Web UI have been attributed to an email address that cannot be covered by necessary legal agreements (metlos@****.noreply). You should amend your commits to fix the email address in the commit. 
See: https://help.github.com/en/articles/changing-a-commit-message and https://help.github.com/en/articles/changing-author-info

**Comment by metlos** [comment]
I've re-authored the commit and force pushed into the branch so that the license checks pass. Thanks for pointing out the problem to me @themr0c.

**Comment by themr0c** [comment]
Thank you for the contribution, and thank you for your hard work fixing the license check.

**Pull request merged**